glambie/processing/process_global_results.py

```diff
@@ -105,32 +109,42 @@ def _combine_regional_results_into_global(regional_results_catalogue: DataCatalo
     Returns
     -------
     Timeseries
-        Global timeseries
+        Globally aggregated timeseries
     """
-
-    # TODO: need to adapt error propagation in this
-
     assert regional_results_catalogue.datasets_are_same_unit()
     assert regional_results_catalogue.datasets[0].unit == "mwe"
 
-    # Multiply by area to calculate area weighted mean
-    regional_results_catalogue = regional_results_catalogue.copy()  # make sure we dont edit existing datasets
-    total_area = 0
-    for ds in regional_results_catalogue.datasets:
-        ds.data.changes = ds.data.changes * ds.region.rgi6_area
-        ds.data.errors = ds.data.changes * ds.region.rgi6_area
-        total_area = total_area + ds.region.rgi6_area
-
-    # combine all regions into global
-    combined_ts, _ = regional_results_catalogue.average_timeseries_in_catalogue(remove_trend=True,
-                                                                                add_trend_after_averaging=True,
-                                                                                out_data_group=GLAMBIE_DATA_GROUPS[
-                                                                                    "consensus"])
-    # divide by area
-    combined_ts.data.changes = combined_ts.data.changes / (total_area / len(regional_results_catalogue.datasets))
-    # PLACEHOLDER FOR ERROR CALC FOR NOW
-    combined_ts.data.errors = combined_ts.data.errors * 0
-
-    combined_ts.region = REGIONS["global"]
-
-    return combined_ts
+    # merge all dataframes
+    catalogue_dfs = [ds.data.as_dataframe() for ds in regional_results_catalogue.datasets]
+    # calculate global area for weighted mean
+    total_area = np.sum([ds.region.rgi6_area for ds in regional_results_catalogue.datasets])
+
+    # multiply changes and errors with area for each region
+    for _, (df, ds) in enumerate(zip(catalogue_dfs, regional_results_catalogue.datasets)):
```

Comment:
No point enumerating if you're not using the index.
```suggestion
    for df, ds in zip(catalogue_dfs, regional_results_catalogue.datasets):
```

Reply:
good point:) done